### ISSUE TITLE
Disabled Button is read twice in JAWS VO mode

### ISSUE THREAD

**Issue by jean182** [microsoft/fluentui#26291]
### Environment Information

- **Package version(s)**: Latest Fluent 8
- **Browser and OS versions**: Chrome and Windows 11.

### Describe the issue:

When a button is disabled and does not have the `allowDisabledFocus` prop set to true use JAWS VO mode (Insert + z while JAWS is being used) and you'll see that the button is announced twice. Oddly enough if the prop `allowDisabledFocus` is set to false the button will be announced just once using VO Mode but the disabled button can be focused and that's something I want to avoid.

#### Actual behavior:

1. Turn VO Mode in JAWS
2. Move around to the disabled button.
3. Verify that moving will announce the button twice (like if there was a hidden button).

#### Expected behavior:

1. Turn VO Mode in JAWS
2. Move around to the disabled button.
3. Verify that moving will announce the buttononly once.

### If applicable, please provide a codepen repro:

https://codepen.io/jean182/pen/jOpBdwO 

It is also repro in the Fluent UI Button examples site https://developer.microsoft.com/en-us/fluentui#/controls/web/button

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No) No

Requested priority: (Blocking, High, Normal, Low) Normal

Products/sites affected: (if applicable) Everywhere when the button is disabled and the `allowDisabledFocus` is set to false
